agent/api/task.go

```diff
@@ -464,6 +464,25 @@ func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerCli
 	return config, nil
 }
 
+// SetConfigHostConfigCommonFileds sets the same fields in both Config and HostConfig for backward compatibility
+func (task *Task) SetConfigHostConfigCommonFileds(container *Container, config *docker.Config, hc *docker.HostConfig) {
+	// Convert MB to B
+	dockerMem := int64(container.Memory * 1024 * 1024)
+	if dockerMem != 0 && dockerMem < DockerContainerMinimumMemoryInBytes {
+		dockerMem = DockerContainerMinimumMemoryInBytes
+	}
+	cpuShare := task.dockerCPUShares(container.CPU)
```

Comment:
Do we really want to set CPU in both?

Reply:
Based on this [commit](https://github.com/moby/moby/commit/837eec064d2d40a4d86acbc6f47fada8263e0d4c), it looks like the memory and cpu are both moved to HostConfig, I will modify this only be set when the api version is <1.18